tests/test_federal_senate_dataset.py

```diff
@@ -1,41 +1,102 @@
 import os
 from tempfile import gettempdir
-from unittest import main, skipIf, TestCase, TestLoader
+from unittest import TestCase
+from unittest.mock import patch
+
+import pandas as pd
 
 from serenata_toolbox.federal_senate.federal_senate_dataset import FederalSenateDataset
 
+
 class TestFederalSenateDataset(TestCase):
-    def setUp(self):
+    @classmethod
+    def setUpClass(cls):
+        cls.expected_files = ['federal-senate-2008.csv',
+                              'federal-senate-2009.csv']
+
+    @patch('serenata_toolbox.federal_senate.federal_senate_dataset.urlretrieve')
+    def test_fetch_files_from_S3(self, mockedUrlRetrieve):
         self.path = gettempdir()
         self.subject = FederalSenateDataset(self.path)
 
-    @skipIf(os.environ.get('RUN_INTEGRATION_TESTS') != '1',
-            'Skipping integration test')
-    def test_fetch_saves_raw_files(self):
-        self.subject.fetch()
-        names = ['federal-senate-{}.csv'.format(year) for year in range(self.subject.FIRST_YEAR, self.subject.NEXT_YEAR)]
-        for name in names:
-            file_path = os.path.join(self.path, name)
-            assert(os.path.exists(file_path))
-
-    @skipIf(os.environ.get('RUN_INTEGRATION_TESTS') != '1',
-            'Skipping integration test')
-    def test_translate_creates_english_versions_for_every_csv(self):
-        self.subject.fetch()
-        self.subject.translate()
-        names = ['federal-senate-{}.xz'.format(year) for year in range(self.subject.FIRST_YEAR, self.subject.NEXT_YEAR)]
-        for name in names:
-            file_path = os.path.join(self.path, name)
-            assert(os.path.exists(file_path))
-
-    @skipIf(os.environ.get('RUN_INTEGRATION_TESTS') != '1',
-            'Skipping integration test')
-    def test_clean_creates_a_reimbursements_file(self):
-        self.subject.fetch()
+        retrieved_files, not_found_files = self.subject.fetch()
+
+        self.assertTrue(mockedUrlRetrieve.called)
+        self.assertEqual(mockedUrlRetrieve.call_count, 10)
```

Comment:
I addressed it recently in this commit: 268148910768afb007b332d460e761dc4a12dfd6
What do you think of this approuch?